### ISSUE TITLE
map.projection.fromLatLngToPoint throws an exception for google_maps >= 3.4.0

### ISSUE THREAD

**Issue by pawsa** [a14n/dart-google-maps#81]
Attempting to use map.projection.fromLatLngToPoint with google_maps versions 3.4.0 and 3.4.2 throws an exception:
```
    final p = map.projection.fromLatLngToPoint(e.latLng, Point(0,0));

Uncaught (in promise) TypeError: Cannot set property 'x' of null
    at _.kg.fromLatLngToPoint (js?key=<KEY>:214)
    at js.JsObject._fromJs.callMethod (dart_sdk.js:51430)
    at google_maps_src.Projection.created.fromLatLngToPoint (google_maps_src.g.dart:2745)
    at page.dart:11
    at _RootZone.runUnaryGuarded (dart_sdk.js:31199)
   ...
```

This code worked fine with google_maps version 3.3.4.

You can reproduce it using the following files `pubspec.yaml`:
```
name: dartgooglemapsprojectionbug
description: demo for the projection bug

environment:
  sdk: '>=2.8.0 <3.0.0'

dependencies:
  google_maps: 3.4.0

dev_dependencies:
  build_runner: ^1.7.2
  build_web_compilers: ">=1.2.0 <3.0.0"
```
`web/index.html`
```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title>dart-google-maps projection bug</title>
    <style>
      html, body, #map-canvas {
        height: 100%;
        margin: 0px;
        padding: 0px
      }
    </style>
  </head>
  <body>
    <div id="map-canvas"></div>

    <script src="http://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY"></script>
    <script defer src="page.dart.js"></script>
  </body>
</html>
```
and `web/page.dart`:
```
import 'dart:html';
import 'package:google_maps/google_maps.dart';

void main() {
  final mapOptions = MapOptions()
    ..zoom = 1
    ..center = LatLng(0, 0);
  final map = GMap(document.getElementById('map-canvas'), mapOptions);

  map.onClick.listen((e) {
      final p1 = map.projection.fromLatLngToPoint(e.latLng, Point(0,0));
      final p2 = map.projection.fromLatLngToPoint(LatLng(0,0), Point(0, 0));
      print('Point.X.Y ${p1}: ${e.latLng} p2: ${p2}');
  });
}
```
You may need to erase your .dart_tool cache between version changes.

**Comment by a14n** [comment]
Thanks for the report.

There's indeed en issue with `fromLatLngToPoint`. The `this` context is lost at some point and it causes this issue. Moreover there's another issue with the `point` optional parameter that don't support passing `null`.

Should be fix with version 3.4.3

**Comment by a14n** [comment]
Closed via 3e3cb4d
